src/Axe/Controllers/CompilerController.cs

```diff
@@ -46,80 +51,64 @@ public IActionResult Solve(int id)
-                    MethodDeclarationSyntax methodDeclaration = SyntaxFactory.MethodDeclaration(SyntaxFactory.ParseTypeName("bool"), "CHECK")
-                        .AddModifiers(SyntaxFactory.Token(SyntaxKind.PublicKeyword))
-                        .WithBody(SyntaxFactory.Block(syntaxOfVerificationCode));
-
-                    ClassDeclarationSyntax axeTaskClass = syntaxTree.GetRoot().DescendantNodes()
-                        .OfType<ClassDeclarationSyntax>()
-                        .First(c => c.Identifier.ValueText == "AxeTask");
-
-                    ClassDeclarationSyntax newAxeTaskClass = axeTaskClass.AddMembers(methodDeclaration);
-                    SyntaxNode syntaxNode = syntaxTree.GetRoot().ReplaceNode(axeTaskClass, newAxeTaskClass);
-                    string sourceCode = syntaxNode.NormalizeWhitespace().ToFullString();
-                    syntaxTree = CSharpSyntaxTree.ParseText(sourceCode);
-
-                    MetadataReference[] references = new MetadataReference[]
-                    {
-                        MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location),
-                        MetadataReference.CreateFromFile(typeof(Hashtable).GetTypeInfo().Assembly.Location)
-                    };
-
-                    CSharpCompilation compilation = CSharpCompilation.Create(
-                        assemblyName: assemblyName,
-                        syntaxTrees: new[] { syntaxTree },
-                        references: references,
-                        options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
-
-                    using (var stream = new MemoryStream())
+                    model.SourceCode = CSharpSyntaxTree.ParseText(model.SourceCode).GetRoot().NormalizeWhitespace().ToFullString();
+                    Tuple<CodeBlockResult, string[]> result = compileManager.Solve(model);
+                    ViewData["StatusOfExecuting"] = result.Item1;
+                    switch (result.Item1)
                     {
-                        EmitResult result = compilation.Emit(stream);
-                        if (!result.Success)
-                        {
-                            IEnumerable<Diagnostic> failures = result.Diagnostics.Where(diagnostic =>
-                                diagnostic.IsWarningAsError ||
-                                diagnostic.Severity == DiagnosticSeverity.Error);
-
-                            foreach (Diagnostic diagnostic in failures)
+                        case CodeBlockResult.SUCCESS:
+                            break;
+                        case CodeBlockResult.FAILED:
+                        case CodeBlockResult.ERROR:
+                            for (int i = 0; i < result.Item2.Length; i++)
                             {
-                                ModelState.AddModelError(diagnostic.Id, diagnostic.GetMessage());
+                                ModelState.AddModelError(i.ToString(), result.Item2[i]);
```

Comment:
pass ModelState to manager with other parameters (in Request<> wrapper) and perform all logic in manager